xpk.py

```diff
@@ -114,6 +114,183 @@
+        completions: {system.vms_per_slice}
+        parallelism: {system.vms_per_slice}
+        template:
+          spec:
+            containers:
+            - args:
+              {pathways_worker_args}
+              env:
+              - name: TPU_STDERR_LOG_LEVEL
+                value: "0"
+              - name: TPU_MIN_LOG_LEVEL
+                value: "0"
+              - name: TF_CPP_MIN_LOG_LEVEL
+                value: "0"
+              - name: XCLOUD_ENVIRONMENT
+                value: GCP
+              image: {args.server_image}
+              imagePullPolicy: Always
+              name: pathways-worker
+              ports:
+              - containerPort: 38677
+              - containerPort: 8471
+              - containerPort: 8080
+              resources:
+                limits:
+                  {resource_type}: {system.chips_per_vm}
+              securityContext:
+                privileged: true
+              volumeMounts:
+              - mountPath: /tmp
+                name: shared-tmp
+            nodeSelector:
+              {accelerator_label}
+              {machine_label}
+            priorityClassName: {args.priority}
+            # hostNetwork: true
+            # dnsPolicy: ClusterFirstWithHostNet
+            volumes:
+            - hostPath:
+                path: /tmp
+                type: DirectoryOrCreate
+              name: shared-tmp
+  - name: rm
+    replicas: 1
+    template:
+      metadata:
+        labels:
+          xpk.google.com/workload: {args.workload}
+      spec:
+        backoffLimit: 0
+        completions: 1
+        parallelism: 1
+        template:
+          spec:
+            containers:
+            - args:
+              {pathways_rm_args}
+              env:
+              - name: TPU_STDERR_LOG_LEVEL
+                value: "0"
+              - name: REPLICATED_JOB_NAME
+                valueFrom:
+                  fieldRef:
+                    fieldPath: metadata.annotations['jobset.sigs.k8s.io/replicatedjob-name']
+              - name: JOBSET_NAME
+                valueFrom:
+                  fieldRef:
+                    fieldPath: metadata.annotations['jobset.sigs.k8s.io/jobset-name']
+              - name: HOST_ADDRESS
+                value: $(JOBSET_NAME)-$(REPLICATED_JOB_NAME)-0-0.$(JOBSET_NAME)
+              - name: TPU_SKIP_MDS_QUERY
+                value: "true"
+              - name: XCLOUD_ENVIRONMENT
+                value: GCP
+              image: {args.server_image}
+              imagePullPolicy: Always
+              name: pathways-rm
+              ports:
+              - containerPort: 38677
+              resources:
+                limits:
+                  cpu: "4"
+                  memory: 8G
+              securityContext:
+                privileged: true
+              volumeMounts:
+              - mountPath: /tmp
+                name: shared-tmp
+            nodeSelector:
+              cloud.google.com/gke-nodepool: cpu-rm-np
```

Comment:
Maybe we need to address this in a separate PR, but GKE now supports running multiple CPU containers (RM, Proxy and USer job) on a single nodepool. Please check with Yash on that

Reply:
Great to know that! Will check how to imbibe it in this/  an upcoming PR

Reply:
In theory this should remove the need for more than one cpu node pool -- aka just have the default cpu node pool. Would be a helpful simplification in the cluster creation for pathways. It might make it easier to do resource management for kueuing of jobs.

Reply:
Hoping to figure this out soon. At the moment, I am having issues having multiple workloads share the CPU nodepools. Issue is observed when one node is occupied with a workload but other nodes are available. This happens for RM, Proxy and User CPU nodepools.

Reply:
Keeping the resource flavors to have the maximum resource limits with autoscaling enabled.